var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -38,78 +38,63 @@ class Mercury(CMakePackage):
     depends_on('bmi', when='+bmi')
     depends_on('mpi', when='+mpi')
     depends_on('libfabric@1.5:', when='+ofi')
-    depends_on('openpa@1.0.3:', when='+opa')
-    depends_on('boost@1.48:', when='+boost')
+    depends_on('openpa@1.0.3:', when='%gcc@:4.8')
+    depends_on('boost@1.48:', when='+boostsys')
+    depends_on('boost', when='@:0.9')  # internal boost headers were added in 1.0.0
+
+    conflicts('+ofi', when='@:0.9')    # libfabric support was added in 1.0.0
+    conflicts('~ofi', when='+udreg')   # udreg option is specific to OFI
 
     def cmake_args(self):
+        """Populate cmake arguments for Mercury."""
         spec = self.spec
-        args = [
-            '-DBUILD_SHARED_LIBS=ON',
+
+        def variant_bool(feature, on='ON', off='OFF'):
```

Comment:
Nothing at all wrong with this, just a suggestion to make it a little more concise and "pythonic" with a simple lambda:
```python
variant_bool = lambda feature : str(feature in self.spec)
```
Since CMake will happily accept `'True'` and `'False'` as `-DFOO:BOOL=` values.

Reply:
thanks for the suggestion! I will make that change